File: dengue_app/lib/models/org_unit.dart
import 'dart:core';

// this class corresponds to the org unit table
class OrgUnit {
  // id is auto incremented in the database
  int id;
  String province;
  String district;
  String name;
  String contact;

  OrgUnit({this.id, this.province, this.district, this.name, this.contact});

  Map toJson() => {
    'id': id,
    'province': province,
    'district': district,
    'name': name,
    'contact': contact,
  };

  factory OrgUnit.fromJson(Map<String, dynamic> json) {
    return OrgUnit(
        id: json['id'] as int,
        province: json['province'] as String,
        district: json['district'] as String,
        name: json['name'] as String,
        contact: json['contact'] as String
    );
  }
}

File: backend/models/org_unit.py
from database import db
from database import ma


class OrgUnit(db.Model):
    # class corresponding to the org_unit Table in the database
    __tablename__ = 'org_unit'
    id = db.Column(db.Integer, primary_key=True)
    province = db.Column(db.String(45), nullable=False)
    district = db.Column(db.String(45), nullable=False)
    name = db.Column(db.String(45), nullable=False)
    contact = db.Column(db.String(10), nullable=False)

    def __init__(self, province, district, name, contact):
        self.province = province
        self.district = district
        self.name = name
        self.contact = contact
    
    def prePopulateOrgUnit():
        # ONLY FOR TESTING PURPOSES
        # adds a dummy org_unit with ID 1 to surpress foreign_key errors
        objects = [
            # status 1
            OrgUnit('Western', 'Colombo', 'National Hosptal of Sri Lanka', '0112691111')
        ]
        # bulk insert operation
        db.session.bulk_save_objects(objects)
        db.session.commit()


class OrgUnitSchema(ma.Schema):
    class Meta:
        fields = ('id', 'province', 'district', 'name', 'contact')


# init schema
org_unit_schema = OrgUnitSchema()
org_units_schema = OrgUnitSchema(many=True)
